language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/inlayhint/InlayHintProvider.java

```diff
@@ -0,0 +1,229 @@
+/*
+ * Copyright (c) 2023, WSO2 LLC. (http://wso2.com) All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.ballerinalang.langserver.inlayhint;
+
+import io.ballerina.compiler.api.symbols.FunctionSymbol;
+import io.ballerina.compiler.api.symbols.FunctionTypeSymbol;
+import io.ballerina.compiler.api.symbols.ModuleSymbol;
+import io.ballerina.compiler.api.symbols.ParameterSymbol;
+import io.ballerina.compiler.api.symbols.Symbol;
+import io.ballerina.compiler.api.symbols.SymbolKind;
+import io.ballerina.compiler.syntax.tree.FunctionCallExpressionNode;
+import io.ballerina.compiler.syntax.tree.MethodCallExpressionNode;
+import io.ballerina.compiler.syntax.tree.Node;
+import io.ballerina.compiler.syntax.tree.NodeVisitor;
+import io.ballerina.compiler.syntax.tree.NonTerminalNode;
+import io.ballerina.compiler.syntax.tree.PositionalArgumentNode;
+import io.ballerina.compiler.syntax.tree.QualifiedNameReferenceNode;
+import io.ballerina.compiler.syntax.tree.SyntaxKind;
+import io.ballerina.tools.text.LineRange;
+import org.ballerinalang.langserver.common.utils.SymbolUtil;
+import org.ballerinalang.langserver.common.utils.TypeResolverUtil;
+import org.ballerinalang.langserver.commons.InlayHintContext;
+import org.eclipse.lsp4j.InlayHint;
+import org.eclipse.lsp4j.Position;
+import org.eclipse.lsp4j.jsonrpc.messages.Either;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+/**
+ * Represents the Inlay Hint Provider.
+ *
+ * @since 2201.7.0
+ */
+public class InlayHintProvider {
+    public static List<InlayHint> getInlayHint(InlayHintContext context) {
+        InlayHintMethodTypeFinder inlayHintMethodFinder = new InlayHintMethodTypeFinder();
+        if (context.currentDocument().isEmpty()) {
+            return Collections.emptyList();
+        }
+        Node rootNode = context.currentDocument().get().syntaxTree().rootNode();
+        rootNode.accept(inlayHintMethodFinder);
+
+        // Get the method name list
+        List<NonTerminalNode> methodNameList = inlayHintMethodFinder.getMethodNameList();
+        List<InlayHint> inlayHints = new ArrayList<>();
+
+        for (NonTerminalNode node : methodNameList) {
+            InlayHintArgumentTypeFinder argumentTypeFinder = new InlayHintArgumentTypeFinder();
```

Comment:
To get the corresponding argument list for each node, we need to create an instance of InlayHintArgumentTypeFinder for each node individually. Otherwise, all the arguments will be added to the same list